app/controllers/concerns/wrap_in_with_deleted.rb

```diff
@@ -3,29 +3,21 @@ module WrapInWithDeleted
   extend ActiveSupport::Concern
 
   included do
-    around_action :wrap_in_with_deleted, if: proc {
-      request.get? && (with_deleted? || search_with_deleted?)
-    }
+    around_action :wrap_in_with_deleted
   end
 
   protected
 
-  def wrap_in_with_deleted
-    Project.with_deleted do
-      Stage.with_deleted do
-        Deploy.with_deleted do
-          yield
-        end
-      end
+  def wrap_in_with_deleted(&block)
+    if request.get? && value = (params[:with_deleted].presence || search_with_deleted?)
+      klasses = value.split(",").map(&:safe_constantize)
```

Comment:
It this is safe?

Reply:
it says "safe_" :trollface: 

users could summon any constant here, but we only call `with_deleted` on it ... so kinda hard to abuse ... we could limit it to AR models 🤷‍♂️